File: src/main/resources/static/app/entities/TransductorValue.js
class TransductorValue {
	
	constructor(property, value, unit, date) {
		this.property = property;
		this.value = value;
		this.unit = unit;
		this.date = date;
	}

	getProperty () {
		return this.property;
	}
	
	getValue() {
		return this.value;
	}

	getValueWithUnit () {
		var value = this.getValue();
		if (this.unit != null) {
			value += ' ' + this.unit;
		}
		return value ;
	}
	
	getDate(){
		return this.date;
	
	}
	
	isDigital () {
		return (this.unit == null);
	}
	
	isHigh () {
		return (this.getValue() == "High");
	}
	
	setValue(value) {
		this.value = value;
	}
	
	static getDigitalStringValue(value) {
		if (value) {
			return "High";
		} else {
			return "Low";
		}
	}
	
	static getDigitalIntValue(value) {
		if (value === "High") {
			return 1;
		} else {
			return 0;
		}
		
	}


}

File: src/main/java/com/iotassistant/models/transductor/TransductorValue.java
package com.iotassistant.models.transductor;

public abstract class TransductorValue {
	
	protected String value;
	
	protected Property property;
	
	protected String date;
	
	public String getValue() {
		return value;
	}

	public String getUnit() {
		return property.getUnit();
	}


	public String getDescriptiveInformationFromValue() {
		return property.getDescriptiveInformationFromValue(value);
	}


	public String getDate() {
		return date;
	}


}
